Binary/Loader/FlysystemLoader.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Liip\ImagineBundle\Binary\Loader;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+use Symfony\Component\HttpFoundation\File\MimeType\ExtensionGuesserInterface;
+use Liip\ImagineBundle\Model\Binary;
+use Liip\ImagineBundle\Exception\Binary\Loader\NotLoadableException;
+
+class FlysystemLoader implements LoaderInterface
+{
+    /**
+     * @var \League\Flysystem\Filesystem
+     */
+    protected $filesystem;
+    
+    /**
+     * @var ExtensionGuesserInterface
+     */
+    protected $extensionGuesser;
+    
+
+    public function __construct(
+        ContainerInterface $container,
+        ExtensionGuesserInterface $extensionGuesser,
+        $sFileSystem)
+    {
+        $this->extensionGuesser = $extensionGuesser;
+        
+        try {
+            $sFileSystemService = sprintf('oneup_flysystem.%s_filesystem', $sFileSystem);
```

Comment:
`oneup_flysystem.%s_filesystem` - looks like it is a service from a bundle. What if someone wants to use this loader with manually created filesystem service?